File: sdks/go/live_test.go
package nrouter

import (
	"context"
	"os"
	"strings"
	"testing"
	"time"
)

// TestLiveClaude is opt-in because it spends real credits. It is the release
// smoke that mock servers cannot replace: the published request shape, native
// Anthropic stream, response headers and local/stage gateway must agree.
func TestLiveClaude(t *testing.T) {
	if os.Getenv("NROUTER_LIVE") != "1" {
		t.Skip("set NROUTER_LIVE=1 to run the billed gateway acceptance")
	}
	baseURL := os.Getenv("NROUTER_BASE_URL")
	if baseURL == "" {
		baseURL = DefaultBaseURL
	}
	client, err := NewFromEnv(WithBaseURL(baseURL))
	if err != nil {
		t.Fatal(err)
	}
	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
	defer cancel()
	body := map[string]any{
		"model":      "claude-3-5-haiku-20241022",
		"max_tokens": 2,
		"messages": []any{
			map[string]any{"role": "user", "content": "Reply OK"},
		},
	}
	stream, err := client.MessagesStream(ctx, body)
	if err != nil {
		t.Fatal(err)
	}
	defer stream.Close()
	var text strings.Builder
	for stream.Next() {
		text.WriteString(stream.Chunk().Delta)
	}
	if err := stream.Err(); err != nil {
		t.Fatal(err)
	}
	if text.Len() == 0 || stream.Meta.RequestID == "" {
		t.Fatalf("live stream lacked text or request metadata: %#v", stream.Meta)
	}
}

File: sdks/dart/test/live_test.dart
import 'dart:io';

import 'package:nrouter/nrouter.dart';
import 'package:test/test.dart';

void main() {
  test(
    'live Claude messages stream returns billing metadata',
    () async {
      final client = NRouter(
        apiKey: Platform.environment['NROUTER_API_KEY']!,
        baseUrl: Platform.environment['NROUTER_BASE_URL'] ??
            'http://127.0.0.1:4000/v1',
      );
      try {
        final chunks = await client.messagesStream({
          'model': 'claude-3-5-haiku-20241022',
          'max_tokens': 2,
          'messages': [
            {'role': 'user', 'content': 'Reply OK'},
          ],
        }).toList();

        expect(chunks, isNotEmpty);
        expect(chunks.first.meta.requestId, isNotEmpty);
        // SSE response headers are committed before final usage is known, so
        // the gateway cannot add the post-stream cost retroactively. Unknown
        // must remain null; rendering it as zero would report a free request.
        expect(chunks.first.meta.isPriced, isFalse);
        expect(chunks.first.meta.cost, isNull);
      } finally {
        client.close();
      }
    },
    skip: Platform.environment['NROUTER_LIVE'] == '1'
        ? false
        : 'set NROUTER_LIVE=1 for the billed local-gateway acceptance',
  );
}
